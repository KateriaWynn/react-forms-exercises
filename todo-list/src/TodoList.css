body {
  padding: 50px;
  background-color: #674172;
  font-family: sans-serif;
}

.todo-list {
  display: flex;
  justify-content: center;
}

input {
  width: 100%;
  background-color: #f1e7fe;
  color: #674172;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

button {
  width: 50%;
  background-color: #d5b8ff;
  color: #674172;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}
