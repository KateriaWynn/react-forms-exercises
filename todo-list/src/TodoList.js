import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import './TodoList.css';

function TodoList() {
  const [todos, setTodos] = useState([]);

  /** Add new todo object to TodoList. */
  const addTodo = (todo) => {
    const newTodo = { ...todo, id: uuid() };
    setTodos((todos) => [...todos, newTodo]);
  };

  /** Remove a todo object from TodoList. */
  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  const todoComponents = todos.map((todo) => (
    <Todo
      key={todo.id}
      id={todo.id}
      task={todo.task}
      handleRemove={removeTodo}
    />
  ));

  return (
    <div className="todo-list">
      <div>
        <NewTodoForm addTodo={addTodo} />
        {todoComponents}
      </div>
    </div>
  );
}

export default TodoList;
