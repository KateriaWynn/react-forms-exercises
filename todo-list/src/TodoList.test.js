import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

// smoke test
test('if TodoList renders without crashing', () => {
  render(<TodoList />);
});

// snapshot test
test('if TodoList matches snapshot', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test('if new todo is added to TodoList', () => {
  const { getByLabelText, queryByText } = render(<TodoList />);
  // no todos yet
  expect(queryByText('Go to post office')).not.toBeInTheDocument();

  const todoInput = getByLabelText('New todo:');
  const submitBtn = queryByText('Add new todo');

  // fill out the form
  fireEvent.change(todoInput, { target: { value: 'Go to post office' } });
  fireEvent.click(submitBtn);

  // a new todo exists!
  expect(queryByText('Go to post office')).toBeInTheDocument();
});

test('if a clicked box is removed from TodoList', () => {
  const { getByLabelText, queryByText } = render(<TodoList />);

  const todoInput = getByLabelText('New todo:');
  const submitBtn = queryByText('Add new todo');

  // fill out the form
  fireEvent.change(todoInput, { target: { value: 'Go to post office' } });
  fireEvent.click(submitBtn);

  // a new todo exists!
  expect(queryByText('Go to post office')).toBeInTheDocument();

  // click the remove button and the todo should be gone
  fireEvent.click(queryByText('X'));
  expect(queryByText('Go to post office')).not.toBeInTheDocument();
});
