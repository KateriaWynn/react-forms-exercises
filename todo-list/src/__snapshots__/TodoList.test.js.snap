// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if TodoList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-list"
  >
    <div>
      <form>
        <label
          for="task"
        >
          New todo: 
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value="Complete React assignments"
        />
        <button>
          Add new todo
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
